src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttributeStore.cs

```diff
@@ -215,23 +215,50 @@ internal bool TryGetPropertyStoreItem(string propertyName, [NotNullWhen(true)] o
                 return _propertyStoreItems.TryGetValue(propertyName, out item);
             }
 
-            private Dictionary<string, PropertyStoreItem> CreatePropertyStoreItems()
+            [RequiresUnreferencedCode("The Type of _type cannot be statically discovered.")]
```

Comment:
It's not `_type` that is the problem. It's the types of its properties.

```suggestion
            [RequiresUnreferencedCode("The Types of _type's properties cannot be statically discovered.")]
```